### PR TITLE
Implement #704, and prevent wobbly on simple click on tittle-bar

### LINKED ISSUE #704: double clicking on decoration doesn't maximize
double clicking on decoration doesn't maximize the window.
works with gtk programs though.

Thks 4 ur attention, interest & support.

### PR DESCRIPTION
**Pull request by Arnaudv6**
Fixes #704

### CODE REVIEW

**Comment by Arnaudv6** [comment]
[handle_action(layout.handle_motion)
](https://github.com/WayfireWM/wayfire/pull/1075/files#diff-e83db7f18f3b3aa510185fd64c195a41035824eb5ce79ccf22395da05b29267fR167) makes for not so intuitive code to read I concede.

File: plugins/decor/deco-layout.cpp [hunk 1]
```diff
@@ -230,7 +243,17 @@ decoration_layout_t::action_response_t decoration_layout_t::handle_press_event(
...
         {
-            return {DECORATION_ACTION_MOVE, 0};
+            if (timer.is_connected())
+            {
+                double_click_at_release = true;
+            } else
+            {
+                timer.set_timeout(300, [=] ()
+                {
+                    timer.disconnect();
```
Comment (ammen99) [inline plugins/decor/deco-layout.cpp:253]: So you say that without this disconnect Wayfire crashes?

**Review by ammen99 — COMMENTED** [review]
I don't understand why you need to disconnect the timer manually, it should just work.
Otherwise, LGTM.

File: plugins/decor/deco-layout.cpp (hunk 1 above)
Comment (Arnaudv6) [reply plugins/decor/deco-layout.cpp:253]: It crashes not, but stays connected: every isolated click after the first one then toggles maximization.

**Comment by Arnaudv6** [comment]
fixes #704. Thanks @ammen99 & @soreau for given assistance

File: plugins/decor/deco-layout.cpp (hunk 1 above)
Comment (ammen99) [reply plugins/decor/deco-layout.cpp:253]: The timer is supposed to automatically disconnect if the handler returns false. There was a bug in core - can you try #1086 and without the  manual disconnect here?

Comment (Arnaudv6) [reply plugins/decor/deco-layout.cpp:253]: yes, tested (on 0.7.0) works fine. Thanks!

**Review by ammen99 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**